language-design.md

```diff
@@ -28,7 +44,10 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
-
+Several factors influence whether or not a language is well-designed. Steele would say that the most important thing is for a language to grow. He says, 
+"If one person does all the work, then growth will be slow. But if one lets the users help do the work, growth can be quick. If many persons work side by side, and the best work is added with care and good taste, a great deal can be added in a short time" [Steele, 1998]. He then gives the example of APL, a language which was well-written and powerful, but very difficult to expand upon; as such, smaller but more easily expandable languages such as Lisp were able to surpass it in time. Thus, Steele would say that the expandability of a language is a main component of good design. I tend to agree with this notion.
+Steele would also way that a well-designed language defines only what is necessary. He says this in his discussion of the Java programming language. He discusses intervals, sets of games, and complex numbers as things he would love to see Java take care of for him, but eventually says he'd rather they not be in the language; he'd prefer the option to add them in as libraries, if needed. And, by extension, those libraries would definitely exist if enough people wanted that capability. The language doesn't have to implement it itself. He says, "When a language gives you the right tools, such classes can be coded by a few and then put up as libraries for other users to use, or not, as they choose; but they don’t have to be built in as part of the base language" [Steele, 1998]. With this in mind, a well-designed language gives programmers the space to add functionality, rather than have it have to be there.
+I think Pavlus would argue that a well designed language is intuitive to learn and read. The experiment on which he reports shows that Perl is more difficult to learn than a randomly designed programming language; implicit in the research is that this makes Perl an inferior language in some way. As Josh discusses in his response to the Pavlus comments, difficulty to learn may increase barriers to entry for a programming language, but doesn't make it worse necessarily or poorly designed. It may be the case that the learning curve is steep, but then that having mastery gives you huge amounts of power in your programming.
 
```

Comment:
@Steele comments: Yes.  Simply put, languages need to be small when they first start out because it's not really possible to create a comprehensive language from scratch.  But we still need large and comprehensive languages.  So, those small languages need to be able to grow.
@Your last paragraph: Great job analyzing Pavlus' argument.  I would've like to see a more explicit agreement/disagreement on your part.  You've pointed out that there are pros (huge power in long term) and cons (steep learning curve) to "difficult" languages like Perl, but which side outweighs the other?